engines/sci/engine/object.h

```diff
@@ -74,21 +76,48 @@ class Object {
 	Object() {
 		_offset = getSciVersion() < SCI_VERSION_1_1 ? 0 : 5;
 		_flags = 0;
-		_baseObj = 0;
-		_baseVars = 0;
+		_baseObj.clear();
+		_baseVars.clear();
 		_methodCount = 0;
-		_propertyOffsetsSci3 = 0;
+		_propertyOffsetsSci3 = nullptr;
 	}
 
 	~Object() {
 		if (getSciVersion() == SCI_VERSION_3) {
-			// FIXME: memory leak! Commented out because of reported heap
-			// corruption by MSVC (e.g. in LSL7, when it starts)
-			//free(_baseVars);
-			//_baseVars = 0;
-			//free(_propertyOffsetsSci3);
-			//_propertyOffsetsSci3 = 0;
+			// TODO: This is super gross
+			free(const_cast<uint16 *>(_baseVars.data()));
+			_baseVars.clear();
+			free(_propertyOffsetsSci3);
+			_propertyOffsetsSci3 = nullptr;
+		}
+	}
+
+	Object &operator=(const Object &other) {
+		_baseObj = other._baseObj;
+		_baseMethod = other._baseMethod;
+		_variables = other._variables;
+		_methodCount = other._methodCount;
+		_flags = other._flags;
+		_offset = other._offset;
+		_pos = other._pos;
+
+		if (getSciVersion() == SCI_VERSION_3) {
```

Comment:
In 9aaf1a008b5e1e75fa6c5b23cc8993a34e0680b9 (SCI32: Fix up Object to use initializers and more containers) and 0aa44aaf85f39295b2de2eb1dde4b7436311eb43 (SCI32: Fix SCI3 save games), `_propertyOffsetsSci3` and `_baseVars` are changed to use `Common::Array`.